Example/CollectionViewToolsExample/Diff/CollectionViewDiffs/CollectionViewDeepDiff.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  Copyright © 2019 Rosberry. All rights reserved.
+//
+
+import DeepDiff
+import CollectionViewTools
+
+final class CollectionViewDeepDiff: CollectionViewDiff {
+
+    func changes<T: DiffItem>(old: [T], new: [T]) -> [CollectionViewChange<T>] {
+        let oldWrappers = old.map { DeepDiffDiffableItemWrapper(item: $0) }
+        let newWrappers = new.map { DeepDiffDiffableItemWrapper(item: $0) }
+        let results = DeepDiff.diff(old: oldWrappers, new: newWrappers)
+        let changes = results.map { result in
+            CollectionViewChange(insert: CollectionViewDeleteInsert(item: result.insert?.item.item,
+                                                                    index: result.insert?.index),
```

Comment:
To avoid this whole mess with optionals, maybe it's better to do something like:
```
let change = CollectionViewChange()
if let insert = result.inset {
    change.insert = CollectionViewDeleteInsert(...)
}
...
```
What do you think?

Reply:
👍